tools/beerocks_analyzer/beerocks_analyzer.py

```diff
@@ -624,129 +628,238 @@ def show():
+            raise ValueError("Path to beerocks_cli not found: {}".format(self.beerocks_cli_path))
+        if not self.target_ip:
+            if self.docker_container_name:
+                try:
+                    # try to find the gateway IP based on the docker network settings
+                    config = json.loads(subprocess.check_output(["docker", "inspect", self.docker_container_name]))[0]
+                    # pick any network:
+                    network_name = next(iter(config["NetworkSettings"]["Networks"]))
+                    network_config = json.loads(subprocess.check_output(["docker", "network", "inspect", network_name]))[0]
+                    self.target_ip = network_config["IPAM"]["Config"][0]["Gateway"]
+                except (KeyError, TypeError):
+                    raise("Could not find the target IP, please use the -my_ip option")
+            else:
+                # running locally:
+                self.target_ip = ip_address("127.0.0.1")
+
+        self.run_flag = True
+
+        command = [str(self.beerocks_cli_path), "-a", str(self.target_ip)]
+        if self.docker_container_name:
+            command = ["docker", "exec", self.docker_container_name] + command
+        self.logger.debug("Starting '%s'", ' '.join(command))
+        self.process = subprocess.Popen(command)
+        outs = None
+        errs = None
+        while self.run_flag:
+            try:
+                outs, errs = self.process.communicate(timeout=1)
+            except subprocess.TimeoutExpired:
+                if outs:
+                    self.logger.info(outs)
+                if errs:
+                    self.logger.error(errs)
+
+    def terminate(self):
+        self.logger.debug("{} terminating".format(self.__class__.__name__))
+        self.run_flag = False
+        if self.process:
+            # note that there is a race between the creation of the Popen object
+            # and its assignment to self.process
+            self.process.kill()
+
+
+class SSHThread(threading.Thread):
+
+    def __init__(self, host="192.168.1.1", my_ip="", ssh_port=22, user='root'):
+        super().__init__()
+        self.logger = logging.getLogger(__name__)
+        self.host = host
+        self.my_ip = my_ip
+        self.ssh_port = ssh_port
+        self.user = user
+        self.ssh_client = paramiko.SSHClient()
+
+    def run(self):
+        self.ssh_client.set_missing_host_key_policy(paramiko.WarningPolicy())
+        self.ssh_client.connect(self.host, port=self.ssh_port, username=self.user)
+
+        if not self.my_ip:
+            self.my_ip = self.ssh_client.get_transport().sock.getsockname()[0]
+
+        self.logger.info("Controller will send update to IP {}".format(self.my_ip))
+        # TODO: add a cli option for the path
+        command = 'LD_LIBRARY_PATH=/opt/beerocks/lib /opt/beerocks/bin/beerocks_cli -a {}'.format(self.my_ip)
+
+        #create ssh shell
+        ssh_shell = self.ssh_client.invoke_shell()
```

Comment:
It's probably possible to tell paramiko to start beerocks_cli directly instead of starting the shell. But then the LD_LIBRARY_PATH issue has to be solved first. (nitpick)